storage/mongo.go

```diff
@@ -157,23 +160,36 @@ func (r *Repository) AddDocument(userID string, ID string) error {
 func (r *Repository) AddVacation(ID string, userID string, startDate int64, durationHours float32) (entities.Vacation, error) {
 	session := r.session.Copy()
 	defer session.Close()
-	var v entities.Vacation
+	vacation := entities.Vacation{
+		ID:            ID,
+		UserID:        userID,
+		StartDate:     startDate,
+		DurationHours: durationHours,
+	}
 	employee, err := r.GetEmployee(userID)
 	if err != nil {
-		return v, err
+		return vacation, err
 	}
 
-	v.ID = ID
-	v.StartDate = startDate
-	v.DurationHours = durationHours
-
 	vacationIndex := "vacations." + strconv.Itoa(len(employee.Vacations))
 	update := make(bson.M)
-	update[vacationIndex] = v
+	update[vacationIndex] = vacation.ID
+
+	if err := session.DB(r.name).C(r.vacations).Insert(vacation); err != nil {
+		return vacation, errors.Wrap(err, "failed to create vacation")
+	}
 
 	if err := session.DB(r.name).C(r.employees).UpdateId(userID, bson.M{"$set": update}); err != nil {
-		return v, err
+		return vacation, errors.Wrap(err, "failed to update employee vacations")
```

Comment:
you can use `fmt.Errorf` with `%w` nowadays. Its a bit ugly I think, but it is now the idiomatic way I believe.

Reply:
And anyway allows to check wrapped errors which is nice.